packages/components/src/DateTimePickers/DateTimePicker/DateTimePicker.component.js

```diff
@@ -1,20 +1,81 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import Immutable from 'immutable';
 import theme from './DateTimePicker.scss';
 import DateTimeView from '../views/DateTimeView';
 import MonthYearView from '../views/MonthYearView';
 
-function DateTimePicker(props) {
-	const isCalendarView = true;
-	const viewComponent = isCalendarView
-		? <DateTimeView />
-		: <MonthYearView />;
-
-	return (
-		<div className={theme.container}>
-			{viewComponent}
-		</div>
-	);
+class DateTimePicker extends React.Component {
+
+	constructor(props) {
+		super(props);
+
+		const now = new Date();
+
+		this.state = {
+			isDateTimeView: true,
+			currentCalendar: Immutable.Map({
+				monthIndex: now.getMonth(),
+				year: now.getFullYear(),
+			}),
+		};
+
+		this.setDateTimeView = this.setView.bind(this, true);
+		this.setMonthYearView = this.setView.bind(this, false);
+		this.onMonthSelected = this.onMonthSelected.bind(this);
+		this.onYearSelected = this.onYearSelected.bind(this);
+	}
+
+	onMonthSelected(monthIndex) {
+		this.setState(previousState => ({
+			currentCalendar: previousState.currentCalendar.merge({ monthIndex }),
+		}));
+	}
+
+	onYearSelected(year) {
+		this.setState(previousState => ({
+			currentCalendar: previousState.currentCalendar.merge({ year }),
+		}));
+	}
+
+	setView(isDateTime) {
+		this.setState({
+			isDateTimeView: isDateTime,
+		});
+	}
+
+	render() {
+		const dateTimeView = (
+			<DateTimeView
+				onTitleClick={this.setMonthYearView}
+				monthIndexSelected={this.state.currentCalendar.get('monthIndex')}
+				yearSelected={this.state.currentCalendar.get('year')}
+				onMonthSelected={this.onMonthSelected}
+				onYearSelected={this.onYearSelected}
```

Comment:
Yes it doesn't feel alright, but there is the month/year used to choose the calendar displayed, and there will be the selected **date** (not just the day).
I've got a hard time to define the naming, because we "select" two "dates", a partial for calendar visual only and the full for the data returned. And the two are not related but they all are selections.

Reply:
&#9888; needs some clear semantic names

Reply:
✓